rt/frontend/jaxrs/src/test/java/org/apache/cxf/jaxrs/impl/ResponseImplTest.java

```diff
@@ -598,6 +601,42 @@ public void testGetLinksMultipleMultiline() {
         }
     }
 
+    @Test
+    public void testReadEntityAndGetEntityAfter() {
+        final String str = "ouch";
+
+        final ResponseImpl response = new ResponseImpl(500, str);
+        final Message outMessage = createMessage();
+        outMessage.put(Message.REQUEST_URI, "http://localhost");
+        response.setOutMessage(outMessage);
+
+        final MultivaluedMap<String, Object> headers = new MetadataMap<>();
+        headers.putSingle("Content-Type", "text/xml");
+        response.addMetadata(headers);
+
+        assertEquals(str, response.readEntity(String.class));
+        assertThrows(IllegalStateException.class, () -> response.getEntity());
```

Comment:
I think that this is an incorrect behavior.  Per the javadocs at https://javadoc.io/static/javax.ws.rs/javax.ws.rs-api/2.1.1/javax/ws/rs/core/Response.html#getEntity-- , the `response.getEntity()` method should not throw an `IllegalStateException` unless the response has been closed or if the entity type is a stream that has been fully consumed.
> Throws:
IllegalStateException - if the entity was previously fully consumed as an input stream, or if the response has been closed.

And I believe that it is expected that the `response.getEntity()` _should_ return the cached entity after calling `readEntity` - see the javadoc from `readEntity(Class)` ( https://javadoc.io/static/javax.ws.rs/javax.ws.rs-api/2.1.1/javax/ws/rs/core/Response.html#readEntity-java.lang.Class- ) :
> A message instance returned from this method will be cached for subsequent retrievals via getEntity(). Unless the supplied entity type is an input stream, this method automatically closes the an unconsumed original response entity data stream if open. In case the entity data has been buffered, the buffer will be reset prior consuming the buffered data to enable subsequent invocations of readEntity(...) methods on this response.

So my take is that the these lines should be:
 ```suggestion
        assertEquals(str, response.readEntity(String.class));
        assertEquals(str, response.getEntity());
        // ideally we would verify that the input stream that backs the entity was closed, but that the Response was not - however in this case the stream is temporary (created via `convertEntityToStreamIfPossible()`)
```

Reply:
Thanks @andymc12 , so this is exactly the way it works right now, but I am actually trying to fix this TCK test (and few others), which verifies this exact flow (unless I miss something): https://github.com/eclipse-ee4j/jakartaee-tck/blob/8.0.2/src/com/sun/ts/tests/jaxrs/ee/rs/core/response/JAXRSClient.java#L343

```
public void getEntityThrowsIllegalStateExceptionWhenConsumedTest()
      throws Fault {
    Response response = invokeGet("entity");
    response.readEntity(String.class);
    try {
      Object entity = response.getEntity();
      fault("No exception has been thrown entity=", entity);
    } catch (IllegalStateException e) {
      logMsg("#getEntity throws IllegalStateException as expected", e);
    }
  }
```